### ISSUE TITLE
Implement byte order fix for lro lamp rdrs

### ISSUE THREAD

**Issue by Sierra-MC** [MillionConcepts/pdr#78]
Endian issue is described for lamp lro in pdr-tests definitions: https://github.com/MillionConcepts/pdr-tests/blob/main/pdr_tests/definitions/lro_lamp/selection_rules.py

Endian gotcha explained here: https://pandas.pydata.org/pandas-docs/version/0.24.0/user_guide/gotchas.html#byte-ordering-issues

Investigate if this is something we are missing that's widespread or just skipped due to special case. Implement fix.

**Comment by m-stclair** [comment]
This is one of many pandas gotchas handled by `np_utils.enforce_order_and_object()` 

The special case must be skipping the part of the table loading workflow that calls this.

**Comment by m-stclair** [comment]
It's less special than that: it's because the products are FITS tables and we don't call that function in the FITS loading workflow -- astropy typically handles this -- so there's something spooky about them as FITS tables. I'll take a look at it.

**Comment by m-stclair** [comment]
This is a more pervasive problem with our FITS binary tables. It was not when we originally implemented that functionality, and I'm guessing there's been some upstream change that's led to it, probably in `astropy`. We probably did not catch it because it is mostly only a problem for printing the dataframe in the interpreter -- which we do not do when we run `dump_browse()`. It also does not affect all such tables, because many of them have nested dtypes, which kicks them to a function that, among other things, calls `enforce_order_and_object`.

It does change the hashes because the dtypes are different, but I bet the hash changes got caught up in some other table-reading change we performed and, again, we didn't notice because the browse products are fine, and tables we spot-checked in memory might not have been affected.

I've added a default byteswap here: 02caa3a54b7cfcb1d6c9a1f3145e525a671e4bac. Spot-checks against binary tables look fine, but we will need to regenerate hashes and may want to test more deeply.
